### PR TITLE
remove scc type carries

### PR DESCRIPTION
**Pull request by deads2k**
I'm getting weird build errors 
```
# github.com/openshift/origin//vendor/github.com/mailru/easyjson/jwriter
vendor/github.com/mailru/easyjson/jwriter/writer.go:10: non-canonical import path "github.com/openshift/origin//vendor/github.com/mailru/easyjson/buffer" (should be "github.com/openshift/origin/vendor/github.com/mailru/easyjson/buffer")
vendor/github.com/mailru/easyjson/jwriter/writer.go:10: can't find import: "github.com/openshift/origin//vendor/github.com/mailru/easyjson/buffer"

````

trying to sort it out.

### CODE REVIEW

File: vendor/k8s.io/kubernetes/pkg/registry/core/rest/scc_patch.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package rest
+
+import (
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+// LegacyStorageWrapper allows someone to mutate the storage.  We use this to add SCCs.
```
Comment (php-coder) [inline vendor/k8s.io/kubernetes/pkg/registry/core/rest/scc_patch.go]: s/LegacyStorageWrapper/LegacyStorageMutatorFunc/

**Comment by deads2k** [comment]
[test]

**Comment by deads2k** [comment]
conflicts!  noooooooooo

**Comment by deads2k** [comment]
[test]

**Comment by deads2k** [comment]
weird, the pull doesn't match https://github.com/deads2k/origin/commits/api-08-scc-addhook

**Comment by deads2k** [comment]
so much for my test :(

**Comment by pweil-** [comment]
@mfojtik fyi

File: pkg/security/api/types.go [hunk 2]
```diff
@@ -6,6 +6,201 @@ import (
...
+	FSTypeFC                    FSType = "fc"
+	FSTypeConfigMap             FSType = "configMap"
+	FSTypeVsphereVolume         FSType = "vsphere"
+	FSTypeQuobyte               FSType = "quobyte"
+	FSTypePhotonPersistentDisk  FSType = "photonPersistentDisk"
+	FSProjected                 FSType = "projected"
+	FSPortworxVolume            FSType = "portworxVolume"
+	FSScaleIO                   FSType = "scaleIO"
+	FSTypeAll                   FSType = "*"
+)
```
Comment (liggitt) [inline pkg/security/api/types.go]: pick in the FSTypeNone bits here

File: pkg/security/api/v1/defaults.go [hunk 3]
```diff
@@ -1,8 +1,80 @@
...
+	}
+	if len(scc.SupplementalGroups.Type) == 0 {
+		scc.SupplementalGroups.Type = SupplementalGroupsStrategyRunAsAny
+	}
+
+	// defaults the volume slice of the SCC.
+	// In order to support old clients the boolean fields will always take precedence.
+	defaultAllowedVolumes := fsTypeToStringSet(scc.Volumes)
+
+	// assume a nil volume slice is allowing everything for backwards compatibility
```
Comment (liggitt) [inline pkg/security/api/v1/defaults.go]: FSTypeNone changes here

File: pkg/security/registry/securitycontextconstraints/BUILD [hunk 4]
```diff
@@ -0,0 +1,49 @@
+package(default_visibility = ["//visibility:public"])
```
Comment (liggitt) [inline pkg/security/registry/securitycontextconstraints/BUILD]: remove BUILD files from the origin packages

File: pkg/security/registry/securitycontextconstraints/doc.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/*
+Copyright 2015 The Kubernetes Authors.
```
Comment (liggitt) [inline pkg/security/registry/securitycontextconstraints/doc.go]: scc files we were carrying should never have had kube copyrights... remove them when moving into origin packages

File: pkg/security/scc/byrestrictions.go [hunk 6]
```diff
@@ -30,44 +30,44 @@ func pointValue(constraint *kapi.SecurityContextConstraints) int {
...
 			hasHostVolume = true
 			// nothing more to do, this is the max point value
 			break
 		// it is easier to specifically list the trivial volumes and allow the
 		// default case to be non-trivial so we don't have to worry about adding
 		// volumes in the future unless they're trivial.
-		case kapi.FSTypeSecret, kapi.FSTypeConfigMap,
-			kapi.FSTypeEmptyDir, kapi.FSTypeDownwardAPI:
+		case securityapi.FSTypeSecret, securityapi.FSTypeConfigMap,
+			securityapi.FSTypeEmptyDir, securityapi.FSTypeDownwardAPI:
```
Comment (liggitt) [inline pkg/security/scc/byrestrictions.go]: FSTypeNone

**Comment by deads2k** [comment]
> FSTypeNone

You merged after this was posted.  How is the structure and after I rebase you can come back through and make sure it was all pickdown down properly.

**Comment by liggitt** [comment]
structure seemed fine, nits overall

**Comment by deads2k** [comment]
```
hack/update-generated-swagger-descriptions.sh
# github.com/openshift/origin/test/extended/util
test/extended/util/test.go:404: c.CoreV1().SecurityContextConstraints undefined (type "github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/client/clientset_generated/clientset/typed/core/v1".CoreV1Interface has no field or method SecurityContextConstraints)
test/extended/util/test.go:417: c.CoreV1().SecurityContextConstraints undefined (type "github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/client/clientset_generated/clientset/typed/core/v1".CoreV1Interface has no field or method SecurityContextConstraints)
Generating fresh completions...
and protobuf.   I hate protobuf
```

**Comment by deads2k** [comment]
I'd like this one before https://github.com/openshift/origin/pull/14843

**Comment by deads2k** [comment]
@stevekuznetsov is there some serial option I can run to see my test-go results?  Locally my machine can't seem to start a built. It constantly fails (even on master).

**Comment by stevekuznetsov** [comment]
@deads2k we should have `scripts/test-go/logs/test-go{,-err}.log` but they don't seem to be present in S3 artifacts -- lucky you.

> Locally my machine can't seem to start a build. It constantly fails (even on master).

Can you:

```
hack/env hack/test-go.sh
```

?

**Comment by deads2k** [comment]
re[test]

**Comment by deads2k** [comment]
re[test]

**Comment by deads2k** [comment]
re[test]

**Comment by deads2k** [comment]
https://github.com/openshift/origin/issues/14043 re[test]

**Comment by deads2k** [comment]
re[test]

**Pull request merged**